service/src/main/java/bio/terra/workspace/common/utils/StepResultWithFlightInfo.java

```diff
@@ -0,0 +1,52 @@
+package bio.terra.workspace.common.utils;
```

Comment:
Is there a case where this is used outside of the context of subflights? If not, it might be best to rename it to indicate that this class only makes sense in the context of subflights. Otherwise, I find the pairing of a step and a flight a bit confusing since it's a hierarchical relationship

Reply:
Actually looking around a bit more, I'm not sure if we need this pairing at all. The steps that call `waitForSubflightCompletionAndReturnFlightInfo` tend to generate their own step results and just rely on the `flightState` included here, which makes it a bit confusing that they're dropping a `StepResult`. The steps that call `waitForSubflightCompletion` do use the step result, but that's all they use. 

if `waitForSubflightCompletionAndReturnFlightInfo` returned `FlightState`, `waitForSubflightCompletion` could then use that return value to determine the appropriate `StepResult` to return.

Reply:
@marctalbott can you take a look at https://github.com/DataBiosphere/terra-workspace-manager/pull/1688/commits/4634c4e87af27af64e370c776baf22a9b0d0c3c4 and let me know what you think? I collapsed the code into a single method, but kept a return object class as it seemed like the cleanest way do the various data/error conversions. But I _think_ that now the relationship between the subflight `FlightState` and `StepStatus` is clearer.

Reply:
I like it, thanks for refactoring that!

Reply:
Great! I'm going to work now on some unittest coverage.